src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1015,3 +1057,924 @@ def upgrade_agents(cmd, client, resource_group_name, cluster_name, kube_config=N
+                           "'az connectedk8s connect -n <connected-cluster-name> -g <resource-group-name>'")
+
+    else:
+        telemetry.set_user_fault()
+        telemetry.set_exception(exception="The azure-arc release namespace couldn't be retrieved", fault_type=consts.Release_Namespace_Not_Found,
+                                summary="The azure-arc release namespace couldn't be retrieved, which implies that the kubernetes cluster has not been onboarded to azure-arc.")
+        raise CLIError("The azure-arc release namespace couldn't be retrieved, which implies that the kubernetes cluster has not been onboarded to azure-arc." +
+                       "Please run 'az connectedk8s connect -n <connected-cluster-name> -g <resource-group-name>' to onboard the cluster")
+    return release_namespace
+
+
+def get_all_helm_values(release_namespace, kube_config, kube_context):
+    cmd_helm_values = ["helm", "get", "values", "--all", "azure-arc", "--namespace", release_namespace]
+    if kube_config:
+        cmd_helm_values.extend(["--kubeconfig", kube_config])
+    if kube_context:
+        cmd_helm_values.extend(["--kube-context", kube_context])
+
+    response_helm_values_get = Popen(cmd_helm_values, stdout=PIPE, stderr=PIPE)
+    output_helm_values, error_helm_get_values = response_helm_values_get.communicate()
+    if response_helm_values_get.returncode != 0:
+        if 'forbidden' in error_helm_get_values.decode("ascii"):
+            telemetry.set_user_fault()
+        telemetry.set_exception(exception=error_helm_get_values.decode("ascii"), fault_type=consts.Get_Helm_Values_Failed,
+                                summary='Error while doing helm get values azure-arc')
+        raise CLIError("Error while getting the helm values in the azure-arc namespace: " + error_helm_get_values.decode("ascii"))
+
+    output_helm_values = output_helm_values.decode("ascii")
+
+    try:
+        existing_values = yaml.safe_load(output_helm_values)
+        return existing_values
+    except Exception as e:
+        telemetry.set_exception(exception=e, fault_type=consts.Helm_Existing_User_Supplied_Value_Get_Fault,
+                                summary='Problem loading the helm existing values')
+        raise CLIError("Problem loading the helm existing values: " + str(e))
+
+
+def enable_features(cmd, client, resource_group_name, cluster_name, features, kube_config=None, kube_context=None,
+                    azrbac_client_id=None, azrbac_client_secret=None, azrbac_skip_authz_check=None):
+    logger.warning("Ensure that you have the latest helm version installed before proceeding.")
+    logger.warning("This operation might take a while...\n")
+
+    features = [x.lower() for x in features]
+    enable_cluster_connect, enable_azure_rbac, enable_cl = utils.check_features_to_update(features)
+
+    if enable_azure_rbac:
+        if (azrbac_client_id is None) and (azrbac_client_secret is None) and (azrbac_skip_authz_check is None):
```

Comment:
>if (azrbac_client_id is None) and (azrbac_client_secret is None) and (azrbac_skip_authz_check is None): [](start = 8, length = 103)

Isnt the check incorrect? we will not error out if client_id is none.. @Anumita

Reply:
since the user can also update the values later (i.e after the first time they update client-id and client-secret) , we can't couple client-id and client-secret